### PR TITLE
Fixes for editor

### PR DESCRIPTION
**Pull request by srclosson**
Acutally inserts selections
code cleanup

### CODE REVIEW

File: src/editor/FluxQueryField.tsx [hunk 1]
```diff
@@ -42,6 +29,13 @@ function expandQuery(bucket, measurement, field) {
   );
 }
 
+class FluxSuggestion extends Suggestion {
```
Comment (davkal) [inline src/editor/FluxQueryField.tsx]: Why is this needed?

File: src/editor/FluxQueryField.tsx [hunk 2]
```diff
@@ -205,16 +199,21 @@ export default class FluxQueryField extends QueryField {
     }
   }, 500);
 
-  applyTypeahead = (editor: CoreEditor, suggestion: { text: any; type: string; deleteBackwards: any }): CoreEditor => {
+  applyTypeahead = (editor: CoreEditor, suggestion: any): CoreEditor => {
     const { typeaheadPrefix, typeaheadContext, typeaheadText } = this.state;
-    let suggestionText = suggestion.text || suggestion;
+
+    if (typeof suggestion['display'] !== 'undefined') {
+      // We have a flux function
```
Comment (davkal) [inline src/editor/FluxQueryField.tsx]: What other suggestions are there?

File: src/editor/FluxQueryField.tsx [hunk 3]
```diff
@@ -249,10 +248,15 @@ export default class FluxQueryField extends QueryField {
         .deleteForward(forward)
         .insertFragment(fragment)
         .moveForward(move)
-        .focus();
+        .focus()
```
Comment (davkal) [inline src/editor/FluxQueryField.tsx]: Any idea why we need to constantly `focus()`?

File: src/editor/QueryField.tsx [hunk 4]
```diff
@@ -222,8 +249,8 @@ class QueryField extends React.Component<any, any> {
     return change || this.state.value.change();
   };
 
-  applyTypeahead = (editor: CoreEditor, suggestion: { text: any; type: string; deleteBackwards: any }): { value: object } => {
-    return { value: new Value() };
+  applyTypeahead = (editor: CoreEditor, suggestion: Suggestion): CoreEditor => {
+    return this.editorEl();
```
Comment (davkal) [inline src/editor/QueryField.tsx]: This function no longer does anything.

File: src/editor/QueryField.tsx [hunk 5]
```diff
@@ -256,7 +283,7 @@ class QueryField extends React.Component<any, any> {
 
   handleClickMenu = (item, editor: CoreEditor) => {
     // Manually triggering change
-    const change = this.applyTypeahead(editor, item);
+    const change = this.applyTypeahead(editor, new Suggestion(item));
```
Comment (davkal) [inline src/editor/QueryField.tsx]: `onChange` takes an editor, should be called that. Also see comment above about `applyTypeahead` not doing much.

File: src/editor/FluxQueryField.tsx (hunk 1 above)
Comment (srclosson) [reply src/editor/FluxQueryField.tsx]: For various reasons:
- I was trying to follow this "override pattern" that I noticed with other plugins. 
- I noticed a difference between a mouse click, and hitting 'tab' or 'enter' on a suggestion. You got fundamentally different objects. This was an attempt to wrap them together

File: src/editor/FluxQueryField.tsx (hunk 2 above)
Comment (srclosson) [reply src/editor/FluxQueryField.tsx]: When you click the mouse, (handleClickMenu). You got a different object than when you used enter and tab. On enter and tab, you got a 'display' field that came from the functions that were described in flux.ts. This was to convert that to the standard suggestion type format.

File: src/editor/FluxQueryField.tsx (hunk 3 above)
Comment (srclosson) [reply src/editor/FluxQueryField.tsx]: None. This was the way it was before and so I didn't question it.

File: src/editor/QueryField.tsx (hunk 4 above)
Comment (srclosson) [reply src/editor/QueryField.tsx]: Nope, and I don't think it did anything before, and is just there to be overrideable so it didn't really matter what was there, accept that I wanted the function declaration and return value to be the same, and so decided to return this.editorEl()

File: src/editor/QueryField.tsx (hunk 5 above)
Comment (srclosson) [reply src/editor/QueryField.tsx]: Yeah, I agree. Let me see if I can clean that one up a bit.

Comment (srclosson) [reply src/editor/QueryField.tsx]: Now that I've looked it through, apply typeahead handles the suggestion and returns the editor. Applytypeahead is overridden, and just needs something as the parent class.

File: src/editor/FluxQueryField.tsx (hunk 2 above)
Comment (davkal) [reply src/editor/FluxQueryField.tsx]: Hmm, can we harmonize this so that mouse and keyboard event handlers turn events into accepted suggestion items?

Comment (briangann) [reply src/editor/FluxQueryField.tsx]: https://github.com/grafana/influxdb-flux-datasource/issues/58

File: src/editor/QueryField.tsx (hunk 5 above)
Comment (briangann) [reply src/editor/QueryField.tsx]: https://github.com/grafana/influxdb-flux-datasource/issues/59
